health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -20,66 +20,116 @@
 import io.stargate.db.datastore.DataStoreFactory;
 import io.stargate.db.datastore.ResultSet;
 import io.stargate.db.datastore.Row;
-import java.util.UUID;
+import io.stargate.db.query.builder.Replication;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableColumn;
+import io.stargate.db.schema.ImmutableTable;
+import io.stargate.db.schema.Table;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
-import org.osgi.framework.BundleContext;
-import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DataStoreHealthChecker extends HealthCheck {
-  private static final Logger logger = LoggerFactory.getLogger(DataStoreHealthChecker.class);
+  private static final Logger logger = LoggerFactory.getLogger(StargateNodesHealthChecker.class);
+  private static final String KEYSPACE_NAME =
+      System.getProperty("stargate.health_check.keyspace_name", "data_store_health_check");
+  private static final String TABLE_NAME =
+      System.getProperty("stargate.health_check.table_name", "health_table");
+  public static final String PK_COLUMN_VALUE = "dummy_pk";
+  public static final String VALUE_COLUMN_VALUE = "dummy_value";
 
-  private final BundleContext context;
+  static final String PK_COLUMN_NAME = "pk";
+  static final String VALUE_COLUMN_NAME = "value";
+  static final Table EXPECTED_TABLE =
+      ImmutableTable.builder()
+          .keyspace(KEYSPACE_NAME)
+          .name(TABLE_NAME)
+          .addColumns(
+              ImmutableColumn.create(PK_COLUMN_NAME, Column.Kind.PartitionKey, Column.Type.Varchar),
+              ImmutableColumn.create(VALUE_COLUMN_NAME, Column.Kind.Regular, Column.Type.Varchar))
+          .build();
+  private final DataStoreFactory dataStoreFactory;
 
-  public DataStoreHealthChecker(BundleContext context) {
-    this.context = context;
+  public DataStoreHealthChecker(DataStoreFactory dataStoreFactory)
+      throws ExecutionException, InterruptedException {
+
+    this.dataStoreFactory = dataStoreFactory;
+    ensureTableExists(dataStoreFactory.createInternal());
```

Comment:
What about making this optional as discussed offline?

Naturally, if this checker is disabled, the `check()` method should probably return perpetual `healthy` state with the appropriate message. WDYT?

Reply:
I was thinking that if the health-check is disabled, we should not register it at all. 
So it can be handled at the `HealthCheckerActivator` level:
```
if(Boolean.parseBoolean(System.getProperty("stargate.health_check.data_store.enabled", "true"))){
         healthCheckRegistry
          .get()
          .register(DATASTORE_CHECK_NAME, new DataStoreHealthChecker(dataStoreFactory.get()));
}
```
wdyt?

Regarding making the `ensureTableExists` optional. Do you want to do it because of the potential auth problem? 
Done in: 
https://github.com/stargate/stargate/pull/742/commits/b5d2072db5042cf95fa430ba09f9867814e91f06
But it means that we have yet another system property parameter.

Reply:
In some cases it may be preferable to create those tables manually. And yes, if DDL forwarding is enabled (assuming DSE backend) the "internal" `DataStore` will not be able to execute DDL (because it does not have a real user, which is required for forwarding DDL).

Reply:
If we do not register the health check sometimes we'd also have to fix `BundleService.defaultHealthCheckNames()` to return the appropriate list, because if a checker is in that list, but not registered, if we cause a failure to be reported by `CheckerResource.checkReadiness()`.

IMHO, it would be cleaner to always register the checker, but not execute the CQL if it is disabled and just return something like `healthy("Storage check disabled")`.

Reply:
done in: https://github.com/stargate/stargate/pull/742/commits/7bab4def792b1660f746f8662dee176c64a449f7